r/version_2/dm06_swap_parent_child_info_v2.R

```diff
@@ -0,0 +1,176 @@
+     # "part_med_risk_v2","part_pregnant", "part_social_group1", "part_social_group2", 
+     # "part_ukitv", "part_workplace_status",
+     ## potentially only in C/D
+     "hhm_covid_contact", "hhm_covid_test", 
+     "hhm_high_risk", "hhm_isolate", "hhm_isolate_atleast_one_day", 
+     "hhm_quarantine", "hhm_quarantine_one_day", "hhm_symp_ache", 
+     "hhm_symp_congestion", "hhm_symp_cough", "hhm_symp_dk", "hhm_symp_fever", 
+     "hhm_symp_no_answer", "hhm_symp_none", "hhm_symp_sob", "hhm_symp_sore_throat", 
+     "hhm_symp_tired"
+   )
+
+for(parent_col in parent_cols) {
+   hhm_col <- gsub("part_", "hhm_", parent_col)
+   dt[mixed_data == T | parent_child == "parent",
+      (hhm_col) := first(get(parent_col)),
+      by = .(part_id, panel, wave, country)]
+   
+}
+
+
+## STEP 5. Fill in child (row_id == child_id) part_data from hhm data
+hhm_cols <- c("hhm_gender", "hhm_age_group")
+
+for(hhm_col in hhm_cols) {
+   part_col <- gsub("hhm_", "part_", hhm_col)
+   dt[mixed_data == T | parent_child == "child", 
+      (part_col) := last(get(hhm_col)), 
+      by = .(part_id, panel, wave, country)]
+}
+
+## STEP 6. Add adult age group
+dt[parent_child == "parent", hhm_age_group := 
+      ifelse(between(hhm_age, 18, 19), "18-19", hhm_age_group)]
+dt[parent_child == "parent", hhm_age_group := 
+      ifelse(between(hhm_age, 20, 24), "20-24", hhm_age_group)]
+dt[parent_child == "parent", hhm_age_group := 
+      ifelse(between(hhm_age, 25, 29), "25-29", hhm_age_group)]
+dt[parent_child == "parent", hhm_age_group := 
+      ifelse(between(hhm_age, 30, 34), "30-34", hhm_age_group)]
+dt[parent_child == "parent", hhm_age_group := 
+      ifelse(between(hhm_age, 35, 39), "35-39", hhm_age_group)]
+dt[parent_child == "parent", hhm_age_group := 
+      ifelse(between(hhm_age, 40, 44), "40-44", hhm_age_group)]
+dt[parent_child == "parent", hhm_age_group := 
+      ifelse(between(hhm_age, 45, 49), "45-49", hhm_age_group)]
+dt[parent_child == "parent", hhm_age_group := 
+      ifelse(between(hhm_age, 50, 54), "50-54", hhm_age_group)]
+dt[parent_child == "parent", hhm_age_group := 
+      ifelse(between(hhm_age, 55, 59), "55-59", hhm_age_group)]
+dt[parent_child == "parent", hhm_age_group := 
+      ifelse(between(hhm_age, 60, 64), "60-64", hhm_age_group)]
+dt[parent_child == "parent", hhm_age_group := 
+      ifelse(between(hhm_age, 65, 69), "65-69", hhm_age_group)]
+dt[parent_child == "parent", hhm_age_group := 
+      ifelse(between(hhm_age, 70, 120), "70+", hhm_age_group)]
+
+
+## STEP 7. Fill in original household size
+dt[sample_type == "child", hh_size := first(hh_size),
+   by = .(country, panel, part_id)]
+
+## STEP 8. Remove now-reduntant child hhm row and assign mixed_data row (row_id == 0) to child 
+dt <- dt[parent_child != "child" | is.na(parent_child)]
+
+dt[mixed_data == TRUE, parent_child := "child"]
+
+
+message(Sys.time() - t)
+
+# For visual testing
```

Comment:
this can be pulled out and used to create testing code later - can remove now if you prefer

Reply:
Leave for now and put somewhere else later